pkg/koordlet/runtimehooks/hooks/resctrl/resctrl.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resctrl
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+
+	apiext "github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/resourceexecutor"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/hooks"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/protocol"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/reconciler"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/rule"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer"
+	util "github.com/koordinator-sh/koordinator/pkg/koordlet/util/resctrl"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util/system"
+	sysutil "github.com/koordinator-sh/koordinator/pkg/koordlet/util/system"
+	rmconfig "github.com/koordinator-sh/koordinator/pkg/runtimeproxy/config"
+)
+
+const (
+	name               = "Resctrl"
+	description        = "set resctrl for pod"
+	ruleNameForAllPods = name + " (AllPods)"
+)
+
+type plugin struct {
+	rule           *Rule
+	engine         util.ResctrlEngine
+	executor       resourceexecutor.ResourceUpdateExecutor
+	statesInformer statesinformer.StatesInformer
+	EventRecorder  record.EventRecorder
+}
+
+var singleton *plugin
+
+func Object() *plugin {
+	if singleton == nil {
+		singleton = newPlugin()
+	}
+	return singleton
+}
+
+func newPlugin() *plugin {
+	return &plugin{
+		rule: newRule(),
+	}
+}
+
+func (p *plugin) Register(op hooks.Options) {
+	// skip if host not support resctrl
+	if support, err := system.IsSupportResctrl(); err != nil {
+		klog.Warningf("check support resctrl failed, err: %s", err)
+		return
+	} else if !support {
+		klog.V(5).Infof("resctrl runtime hook skipped, cpu not support CAT/MBA")
+		return
+	}
+
+	if vendorID, err := sysutil.GetVendorIDByCPUInfo(sysutil.GetCPUInfoPath()); err == nil {
```

Comment:
In another case, the CPU model supports the RDT but the resctrl fs is not mounted. We should find a place to handle it.

Reply:
OK, let's check the mounted status here.